pandas/tests/groupby/test_groupby.py

```diff
@@ -2311,6 +2311,15 @@ def test_groupby_series_with_tuple_name():
     tm.assert_series_equal(result, expected)
 
 
+def test_time_series_groupby_series_named_with_tuple():
+    # GH 42731
+    ser = Series([1, 2, 3, 4], index=[Timestamp(2021, 7, 28 + i) for i in range(4)])
+    grp = Series([1, 0, 1, 0], name=('a', 'a'))
+    result = ser.groupby(grp).last()
+    expected = ser.groupby(grp.rename(None)).last()
```

Comment:
Passing a Series into the groupby will try to align, e.g.

```
ser = Series([1, 2, 3, 4], index=[3, 1, 2, 0])
grp = Series([1, 0, 1, 0])
print(ser.groupby(grp).sum())
```

gives the result

```
0    3
1    7
dtype: int64
```

In this case - the index won't align and you end up with an empty result, is that right?